numba/types.py

```diff
@@ -714,6 +715,33 @@ def is_contig(self):
         return self.layout in 'CF'
 
 
+class VoidArray(Array):
+    """
+    A VoidArray is an array nested within a structured type (which are "void"
```

Comment:
OK - renamed to `NestedArray`... I had been trying to keep consistency with NumPy, but I agree that that actually makes things less clear.